fs_test.go: Assume non-root user instead of failing the test.

pmacik opened the pull request:
When the unit tests are run in a container as a random-like user ID (e.g. in Kubernetes or OpenShift) or the user ID is not mentioned in `/etc/passwd` the tests fail complaining like:
```
--- FAIL: TestCopyDirFail_SrcInaccessible (0.00s)
    fs_test.go:181: Failed to get name of current user: user: unknown userid 1026330000
--- FAIL: TestCopyDirFail_DstInaccessible (0.00s)
    fs_test.go:220: Failed to get name of current user: user: unknown userid 1026330000
--- FAIL: TestCopyDirFailOpen (0.00s)
    fs_test.go:320: Failed to get name of current user: user: unknown userid 1026330000
--- FAIL: TestCopyFileFail (0.00s)
    fs_test.go:489: Failed to get name of current user: user: unknown userid 1026330000
--- FAIL: TestIsDir (0.00s)
    fs_test.go:580: Failed to get name of current user: user: unknown userid 1026330000
--- FAIL: TestIsSymlink (0.00s)
    fs_test.go:637: Failed to get name of current user: user: unknown userid 1026330000
```

The `currentUser` here is really only used to check if it is `root` and the very same function that is used here to get the current user (`user.Current()`) is not used anywhere else in the codebase. It should be sufficient to try to get the current user and if that fails, assume the user is a non-root and go on without actually failing the unit test.

This makes the unit tests less flaky.

technosophos commented:
What is the process, @bacongobbler or @mattfarina, for patching tests in external (but forked) libraries like this? We normally try to stay as close to upstream, right?

sbose78 commented:
Hi @mattfarina and @adamreese Would you be able to take a look at this ? We are blocked with running these tests in OpenSift :-(

sbose78 commented:
@mattfarina @bacongobbler @waveywaves @adamreese Need help with this :)

technosophos reviewed (approved):
Alright, I just trawled through the commit history for this package, and it appears that our fork is a rather permanent feature of Helm at this point (though I wish this were not the case). 

Is there a reason why we are testing against a string name? Wouldn't it be better to test against the UID itself? I am concerned that we might mask a legitimate error with the proposed PR.

technosophos reviewed (commented):
Oops... sorry... meant to mark this as "Comment" not "Approved"

pedjak committed:
fs_test: use os.Getuid() instead user.Current() to determine if a test is executed with root privileges.

This change lower the expectations on test env setup, i.e. tests could be executed in a container under a random UID,
without require an user in /etc/passwd

Signed-off-by: Predrag Knezevic <pknezevi@redhat.com>

pedjak commented:
@technosophos thanks for the hint - the PR is updated to use `os.Getuid()` instead of `user.Current()`. Please take a look.

technosophos reviewed (approved):
Ooo! I very much like this new version!

Alright, given that it is my understanding that we (the Helm Core Maintainers) have made a long-term commitment to support this as a permanent fork of the (now abandoned) upstream, I am LGTMing this.

mattfarina reviewed (approved):
The change has the same intent. lgtm
